pkg/virtual/syncer/coordination/helpers.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coordination
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	syncercontext "github.com/kcp-dev/kcp/pkg/virtual/syncer/context"
+	"github.com/kcp-dev/kcp/pkg/virtual/syncer/transformations"
+)
+
+func FilteredSyncerViewsChanged(old, new metav1.Object, keepSyncTarget func(syncTargetKey string) bool) bool {
+	oldSyncerViewAnnotations := make(map[string]string)
+	for name, value := range old.GetAnnotations() {
+		if strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			if keepSyncTarget(strings.TrimPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix)) {
+				oldSyncerViewAnnotations[name] = value
+			}
+		}
+	}
+
+	newSyncerViewAnnotations := make(map[string]string)
+	for name, value := range new.GetAnnotations() {
+		if strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			if keepSyncTarget(strings.TrimPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix)) {
+				newSyncerViewAnnotations[name] = value
+			}
+		}
+	}
+	return !reflect.DeepEqual(oldSyncerViewAnnotations, newSyncerViewAnnotations)
+}
+
+func SyncerViewsChanged(old, new metav1.Object) bool {
```

Comment:
doc

Reply:
So easy to confuse with the next function. How about `AnySyncerViewChanged`?

Reply:
Renamed